backend/business-partner-agent/src/main/java/org/hyperledger/bpa/impl/aries/ProofTemplateConversion.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2020-2021 - for information on the respective copyright owner
+ * see the NOTICE file and/or the repository at
+ * https://github.com/hyperledger-labs/business-partner-agent
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.hyperledger.bpa.impl.aries;
+
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import org.hyperledger.aries.api.present_proof.PresentProofRequest;
+import org.hyperledger.bpa.api.exception.PartnerException;
+import org.hyperledger.bpa.impl.prooftemplates.*;
+import org.hyperledger.bpa.model.BPAProofTemplate;
+import org.hyperledger.bpa.model.Partner;
+import org.hyperledger.bpa.repository.PartnerRepository;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import javax.validation.Valid;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.function.UnaryOperator;
+
+@Slf4j
+@Singleton
+public class ProofTemplateConversion {
+
+    @Inject
+    PartnerRepository partnerRepo;
+
+    @Inject
+    ProofTemplateConditionOperators<PresentProofRequest.ProofRequest.ProofRequestedPredicates.ProofRequestedPredicatesBuilder, PresentProofRequest.ProofRequest.ProofRequestedAttributes.ProofRequestedAttributesBuilder, PresentProofRequest.ProofRequest.ProofRestrictions.ProofRestrictionsBuilder> ariesConditionOperators;
```

Comment:
Begin --->
I trust that this does whatever it does.

Reply:
This class I have to test thoroughly.